dotnet/WPF/VisualLayerIntegration/VisualLayerIntegration/BarGraphUtility/BarGraph.cs

```diff
@@ -394,61 +416,39 @@ public void UpdateGraphData(string title, string xAxisTitle, string yAxisTitle,
             }
 
             // Reset to new data.
-            graphData = newData;
+            _graphData = newData;
         }
 
-        private void AddLight()
+        private void UpdateLightTargets()
         {
-            ambientLight = compositor.CreateAmbientLight();
-            ambientLight.Color = Colors.White;
-            ambientLight.Targets.Add(mainContainer);
-
-            var innerConeColor = Colors.White;
-            var outerConeColor = Colors.AntiqueWhite;
-
-            barOutlineLight = compositor.CreateSpotLight();
-            barOutlineLight.InnerConeColor = innerConeColor;
-            barOutlineLight.OuterConeColor = outerConeColor;
-            barOutlineLight.CoordinateSpace = mainContainer;
-            barOutlineLight.InnerConeAngleInDegrees = 45;
-            barOutlineLight.OuterConeAngleInDegrees = 80;
-
-            barOutlineLight.Offset = new System.Numerics.Vector3(0, 0, 80);
-
             // Target bars outlines with light.
-            for (int i = 0; i < barValueMap.Count; i++)
+            _barOutlineLight.Targets.RemoveAll();
+            for (int i = 0; i < _bars.Length; i++)
             {
-                var bar = (Bar)barValueMap[i];
-                barOutlineLight.Targets.Add(bar.OutlineRoot);
+                var bar = (Bar)_bars[i];
+                _barOutlineLight.Targets.Add(bar.OutlineRoot);
             }
 
-
-            barLight = compositor.CreatePointLight();
-            barLight.Color = outerConeColor;
-            barLight.CoordinateSpace = mainContainer;
-            barLight.Intensity = 0.5f;
-
-            barLight.Offset = new System.Numerics.Vector3(0, 0, 120);
-
             // Target bars with softer point light.
-            for (int i = 0; i < barValueMap.Count; i++)
+            _barLight.Targets.RemoveAll();
+            for (int i = 0; i < _bars.Length; i++)
             {
-                var bar = (Bar)barValueMap[i];
-                barLight.Targets.Add(bar.Root);
+                var bar = _bars[i];
+                _barLight.Targets.Add(bar.Root);
             }
         }
 
         public void UpdateLight(SysWin.Point relativePoint)
         {
-            barOutlineLight.Offset = new System.Numerics.Vector3((float)relativePoint.X,
-                (float)relativePoint.Y, barOutlineLight.Offset.Z);
-            barLight.Offset = new System.Numerics.Vector3((float)relativePoint.X,
-                (float)relativePoint.Y, barLight.Offset.Z);
+            _barOutlineLight.Offset = new System.Numerics.Vector3((float)relativePoint.X,
+                (float)relativePoint.Y, _barOutlineLight.Offset.Z);
+            _barLight.Offset = new System.Numerics.Vector3((float)relativePoint.X,
+                (float)relativePoint.Y, _barLight.Offset.Z);
         }
 
-        private float GetMaxBarValue(float[] data)
+        private double GetMaxBarValue(double[] data)
         {
-            float max = data[0];
+            double max = data[0];
```

Comment:
Instead of implementing a max function, just use Enumerable.Max(data).

Reply:
updated